public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -268,7 +273,13 @@ export function createDashboardSceneFromDashboardModel(oldModel: DashboardModel,
       new behaviors.LiveNowTimer({ enabled: oldModel.liveNow }),
       preserveDashboardSceneStateInLocalStorage,
       new ScopesFacade({
-        handler: (facade) => sceneGraph.getTimeRange(facade).onRefresh(),
+        handler: (facade) => {
+          if (reloadOnScopesChange && reloadCb && oldModel.uid) {
+            reloadCb();
```

Comment:
not sure I like this reload callback or handler. 

A cleaner thing might be for DashboardScenePageStateManager to detect the option "reloadOnScopesChange" and subscribe to time range and reload. Or for ScopesFacade to emit an event that DashboardScenePageStateManager subscribes to (ReloadDashboardFromAPIEvent or something), we maybe have it. 

Another option is to use locationService.reload()  (which uses the routeReloadCounter)

Reply:
The way we reload right not is more of a "let's see if there are any side effects if we just reload the dashboard like this".

I don't know if making `DashboardScenePageStateManager` responsible for the actual subscription would not cause side effects as it is a singleton that, AFAIK, doesn't keep track of if and what dashboard is loaded.

I'll try `locationService.reload` and see how it goes. That'll allow us to skip passing the callback.

Reply:
Moved to `locationService.reload` which works fine. Thanks for the suggestion!